src/muxer/mplex/mplex_stream.cpp

```diff
@@ -40,10 +40,32 @@ namespace libp2p::connection {
     read(out, bytes, std::move(cb), true);
   }
 
+  void MplexStream::readDone(outcome::result<size_t> res) {
+    auto cb{std::move(reading_->cb)};
+    reading_.reset();
+    cb(res);
+  }
+
+  bool MplexStream::readTry() {
+    auto size{std::min(read_buffer_.size(), reading_->bytes)};
+    if (reading_->some ? size == 0 : size != reading_->bytes) {
```

Comment:
Oof, maybe extract it to a variable with a name that describes what's going on here? It takes a few seconds to parse this expression.